app/controllers/invitations_controller.rb

```diff
@@ -1,114 +1,29 @@
 class InvitationsController < ApplicationController
-  include InvitationsHelper
-  before_filter :load_invitable, only: [:new, :create]
-  before_filter :authenticate_user!, only: :create
+  include PrettyUrlHelper
 
-  rescue_from ActiveRecord::RecordNotFound do
-    render 'application/display_error',
-      locals: { message: t(:'invitation.invitation_not_found') }
-  end
-
-  rescue_from Invitation::InvitationCancelled do
-    render 'application/display_error',
-      locals: { message: t(:'invitation.invitation_cancelled') }
-  end
-
-  rescue_from Invitation::InvitationAlreadyUsed do
-    if current_user and @invitation.accepted?
-      redirect_to @invitation.invitable
+  rescue_from(ActiveRecord::RecordNotFound)    { respond_with_error :"invitation.invitation_not_found" }
+  rescue_from(Invitation::InvitationCancelled) { respond_with_error :"invitation.invitation_cancelled" }
+  rescue_from(Invitation::InvitationAlreadyUsed) do
```

Comment:
No, I just have the happy paths; I'll write some for the unhappy paths. I've had a really great time writing some e2ez around this stuff